### PR TITLE
Update with latest surface fixes based Java export review.

### PR DESCRIPTION
**Pull request by shinfan**
Refactorings:
- Remove innerClass and move util methods to top level.
- Rename ApiCallableBuilder to just Builder
- Rename XPath into XName
- Rename XMethod into XCallSettings in XSettings classe

Improvements:
- Let the user decide whether to auto-close provided channel
- Make all generated methods final
- Use Duration instead of long to avoid time unit confusion
- Support bundling limits configuration

Others:
- Update unit tests
- Update pom and dependency

### CODE REVIEW

**Comment by aozarov** [comment]
Hey Shin per earlier comment do you mind to update this branch from master or at least update travis config so I can tell that this build is not broken?

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java [hunk 1]
```diff
@@ -84,97 +80,80 @@
...
-     * <!-- end manual edit -->
-     */
-    public static final String formatProjectPath(String project) {
-      return PROJECT_PATH_TEMPLATE.instantiate("project", project);
-    }
+  /**
+   * A PathTemplate representing the fully-qualified path to represent
+   * a project resource.
+   *
+   * <!-- manual edit -->
```
Comment (aozarov) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java]: Do you see any reason for providing this '` for a 'private' field?

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java [hunk 2]
```diff
@@ -84,97 +80,80 @@
...
-      return PROJECT_PATH_TEMPLATE.parse(projectPath).get("project");
-    }
+  /**
+   * Formats a string containing the fully-qualified path to represent
+   * a project resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String formatProjectName(String project) {
```
Comment (aozarov) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java]: Do you see any reason for providing this '` for a 'final' method?

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java [hunk 3]
```diff
@@ -84,97 +80,80 @@
...
+    return PROJECT_PATH_TEMPLATE.instantiate("project", project);
```
Comment (aozarov) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java]: did you consider `format`? [e.g.](https://docs.oracle.com/javase/7/docs/api/java/text/MessageFormat.html#format(java.lang.String,%20java.lang.Object...)).

Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java]: The path template string that we get as input, which is language independent and relates to the definition of the API in the proto, supports a different set of things than what MessageFormat supports: https://github.com/googleapis/gax-java/blob/master/src/main/java/com/google/api/gax/protobuf/PathTemplate.java

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java (hunk 2 above)
Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java]: The manual edit section is for adding documentation that is relevant to this Java entry point that the code generator can't add in the general case. This is independent of whether the method is final or not.

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java]: Well, in fact, we probably don't need a comment on this at all.

File: gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java (hunk 3 above)
Comment (aozarov) [inline gcloud-java-pubsub/baseline/src/main/java/com/google/gcloud/pubsub/spi/v1/PublisherApi.java]: No doubt, but don't you think `format` would be more appropriate (and consistent with the parse as well as your own javadoc "Formats a string containing the...."?

**Pull request closed without merging**